pkgs/os-specific/linux/kernel/common-flags.nix

```diff
@@ -0,0 +1,44 @@
+{
+  lib,
+  stdenv,
+  buildPackages,
+  extraMakeFlags ? [ ],
+}:
+# Absolute paths for compilers avoid any PATH-clobbering issues.
+[
+  "CC=${
+    if stdenv.cc.isClang then
+      lib.getExe' stdenv.cc.cc "clang"
```

Comment:
I was about to ask what this is about then I found https://github.com/NixOS/nixpkgs/pull/390631/commits/fbc5923fb30c7e1957a729f19f22968083fb473f.

Can you add the reasoning as comment here, please? Because without the necessary context the condition here looks pretty weird :D

Reply:
Done

Reply:
Why only for clang?

Reply:
I think it was to get around the warnings and #321667

Reply:
I just realized that this might be a bit difficult if we don't have a `meta.mainProgram` for every unwrapped CC. It is because we'd have to duplicate the CC detection logic from the CC wrapper.

Reply:
What's the problem with requiring every C compiler to have a mainProgram set?

Reply:
Done

Reply:
I'm not sure because at least for me, I want to use LLVM's linker. If we make it unconditional then I'd be forced to use GNU's linker. We set it to work around a problem with the linker wrapper. Would just using the unwrapped linker be fine?

Reply:
Using the unwrapped linker was what I meant.